src/pages/gl-system-params/form/SystemParamsForm.js

```diff
@@ -0,0 +1,370 @@
+        myObject[obj.key] = obj.value === 'true' || obj.value === true
+      } else if (
+        obj.key === 'GLDOESeg0End' ||
+        obj.key === 'GLDOESeg0Start' ||
+        obj.key === 'GLFYCSeg0Start' ||
+        obj.key === 'GLFYCSeg0End'
+      ) {
+        myObject[obj.key] = obj.value || null
+      } else {
+        myObject[obj.key] = obj.value ? parseInt(obj.value, 10) : null
+      }
+    })
+
+    formik.setValues(myObject)
+
+    const accountMappings = [
+      { key: 'GLDOEGainAccountId', refField: 'GLDOEGainAccountref', nameField: 'GLDOEGainAccountname' },
+      { key: 'GLDOELossAccountId', refField: 'GLDOELossAccountref', nameField: 'GLDOELossAccountname' },
+      { key: 'GLRoundingAccountDb', refField: 'GLRoundingAccountDbref', nameField: 'GLRoundingAccountDbname' },
+      { key: 'GLRoundingAccountCr', refField: 'GLRoundingAccountCrref', nameField: 'GLRoundingAccountCrname' },
+      { key: 'GLFYCGainAccountId', refField: 'GLFYCGainAccountref', nameField: 'GLFYCGainAccountname' },
+      { key: 'GLFYCLossAccountId', refField: 'GLFYCLossAccountref', nameField: 'GLFYCLossAccountname' }
+    ]
+
+    await Promise.all(
+      accountMappings.map(async ({ key, refField, nameField }) => {
+        const response = await getRequest({
+          extension: GeneralLedgerRepository.ChartOfAccounts.get,
+          parameters: `_recordId=${myObject[key]}`
+        })
+        formik.setFieldValue(refField, response.record.accountRef)
+        formik.setFieldValue(nameField, response.record.name)
+      })
+    )
+  }
+
+  const postSystemParams = obj => {
+    var data = []
+    Object.entries(obj).forEach(([key, value]) => {
+      const newObj = { key: key, value: value }
+      data.push(newObj)
+    })
+    postRequest({
+      extension: SystemRepository.Defaults.set,
+      record: JSON.stringify({ SysDefaults: data })
+    }).then(res => {
+      if (res) toast.success(platformLabels.Edited)
+      updateDefaults(data)
+    })
+  }
+
+  return (
+    <FormShell
+      resourceId={ResourceIds.SystemParams}
+      form={formik}
+      maxAccess={access}
+      infoVisible={false}
+      isSavedClear={false}
+      isCleared={false}
+    >
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
+            <Grid item xs={12}>
+              <ResourceLookup
+                endpointId={GeneralLedgerRepository.ChartOfAccounts.snapshot}
+                valueField='accountRef'
+                displayField='name'
+                name='GLDOEGainAccountId'
+                label={_labels.glDoeGain}
+                form={formik}
+                required
+                displayFieldWidth={2}
+                valueShow='GLDOEGainAccountref'
+                secondValueShow='GLDOEGainAccountname'
+                maxAccess={access}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('GLDOEGainAccountId', newValue ? newValue.recordId : '')
```

Comment:
this structure is repetitive, there are multiple in this single file, you can create a custom onChange something like 
` const handleChange = fieldArray => {
    fieldArray.forEach(({ name, value }) => {
      formik.setFieldValue(name, value)
    })
  }`

and inside the onchange it could be something like 
`const { recordId = '', accountRef = '', name = '' } = newValue || {};
handleChange([{name:'GLDOEGainAccountId',value:recordId},{name:'GLDOEGainAccountref',value:accountRef},{name:'GLDOEGainAccountname',value:name}])
`

it would be more readable and more maintainable and not very repetitive